File: 문민우/프로그래머스/level2/다음큰숫자.java
package level2;

public class 다음큰숫자 {
    public int solution(int n) {

        int answer = 0;

        String binaryStr = Integer.toBinaryString(n);  // 2진수 변환
        int count = 0;

        for(int i = 0; i < binaryStr.length(); i++) {

            if(binaryStr.charAt(i) == '1')  // 1의 개수 카운트
                count++;
        }

        while(true) {  // 다음 수를 2진수로 변환한 뒤, 1의 개수 카운트 반복

            n++;
            String binaryToCheck = Integer.toBinaryString(n);

            int check = 0;

            for(int i = 0; i < binaryToCheck.length(); i++) {

                if(binaryToCheck.charAt(i) == '1')
                    check++;
            }

            if(check == count) {
                answer = Integer.parseInt(binaryToCheck, 2);  // 2진수 -> 10진수
                break;
            }
        }

        return answer;
    }
}

File: 이솔/프로그래머스/Level2/다음 큰 숫자.py
def solution(n):
#     count=0
#     num=0
#     arr=[]
#     k=n
    
#     while n>0:
#         a=n%2
#         n=(n-a)//2
        
#         arr.append(a)
#         if a==1:
#             k+=1
    
#     for i in range(0,len(arr)):
#         num+=arr[i]*(10**(count))
#         count+=1
        
    for i in range(n+1,1000000):
        if bin(n).count('1')==bin(i).count('1'):
            answer=i
            break    
    
    return answer